misk-redis/src/main/kotlin/misk/redis/RedisModule.kt

```diff
@@ -35,10 +37,19 @@ class RedisModule(
     )
 
     // Authenticate with the redis server
-    jedisPool.resource.use {
-      it.auth(replicationGroup.redis_auth_password)
+    try {
+      jedisPool.resource.use { it.auth(replicationGroup.redis_auth_password) }
+    } catch(e: JedisConnectionException) {
+      log.error { "Failed to connect to redis cluster: " +
```

Comment:
an alternative approach is to throw a new exception that holds the context you need. that way you don't have a separate error message that is unconnected to the final stack trace.

```
class RedisConnectionException : RuntimeException(ReplicationGroup, Cause) {
  super("Failed to connect...", cause);
}

catch (e: JedisConnectionException) {
  throw RedisConnectionException(replicationGroup, e);
}
```

Reply:
Nice I like that, I'll switch to creating a new exception